### PR TITLE
SInatra app from rubizza code camp

### PR DESCRIPTION
**Pull request by mesmerze**

### CODE REVIEW

File: jwt.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  def protected!
+    return if authorized?
+    redirect to('/login')
+  end
+
+  # helper to extract the token from the session, header or request param
+  # if we are building an api, we would obviously want
+  # to handle header or request param
+  def extract_token
+    # check for the access_token header
```
Comment (nplusp) [inline jwt.rb]: Why you don't use `||` here? 
```ruby
request.env['access_token'] || request['access_token'] || session['access_token']
```

File: Gemfile~ [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+gem 'heroku'
+gem 'oauth2'
+gem 'pg', '~> 0.18.4'
+gem 'sinatra'
+gem 'sinatra-activerecord', :require => 'sinatra/activerecord'
+gem 'vk', '~> 0.6.5.53'
+gem 'jwt', '~> 1.5', '>= 1.5.6'
+gem 'thin', '~> 1.7', '>= 1.7.2'
+gem 'heroku', '~> 3.99'
+gem 'rake', '~> 12.0'
```
Comment (serg-kovalev) [inline Gemfile~]: this is "bak" file. It had to be added to `.gitignore`
Please delete it from repo and improve `.gitignore` file

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+get('/login') do
+  if params[:code]
+    code = params[:code]
+    uri = URI('https://oauth.vk.com/access_token')
+    shmarams = { client_id: ENV['APP_ID'],
+                 client_secret: ENV['APP_SECRET'],
+                 redirect_uri: "#{ENV['APP_LOC']}/login",
+                 code: code }
+    uri.query = URI.encode_www_form(shmarams)
+    http = Net::HTTP.get_response(uri)
```
Comment (serg-kovalev) [inline app.rb]: what if I get any errors there?
you should not rely on 3rd party services. You need to improve app logic to deal with exceptions.

File: app.rsa [hunk 4]
```diff
@@ -0,0 +1,27 @@
+-----BEGIN RSA PRIVATE KEY-----
```
Comment (serg-kovalev) [inline app.rsa]: remove it from repo. It should be in `.gitignore`

File: app.rsa.pub [hunk 5]
```diff
@@ -0,0 +1,9 @@
+-----BEGIN PUBLIC KEY-----
```
Comment (serg-kovalev) [inline app.rsa.pub]: remove it from repo. It should be in `.gitignore`

File: config/environments.rb [hunk 6]
```diff
@@ -0,0 +1,13 @@
+configure :production, :development do
+data = YAML.load_file('config/database.yml')['development']
+db = OpenStruct.new(data)
+
+    ActiveRecord::Base.establish_connection(
```
Comment (serg-kovalev) [inline config/environments.rb]: style guide
btw, you can use it like `ActiveRecord::Base.establish_connection db`

File: db/migrate/20170711151901_create_user.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+class CreateUser < ActiveRecord::Migration[5.1]
+  def up
+  	create_table :users do |t|
```
Comment (serg-kovalev) [inline db/migrate/20170711151901_create_user.rb]: style guide
check styles everywhere please and setup your editor/ide correctly

File: README.MD [hunk 8]
```diff
@@ -0,0 +1 @@
+shawarmas app
```
Comment (serg-kovalev) [inline README.MD]: where is the link to your app at heroku?

File: app.rb [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+    uri = URI('https://oauth.vk.com/access_token')
+    shmarams = { client_id: ENV['APP_ID'],
+                 client_secret: ENV['APP_SECRET'],
+                 redirect_uri: "#{ENV['APP_LOC']}/login",
+                 code: code }
+    uri.query = URI.encode_www_form(shmarams)
+    http = vk_unavailable(uri)
+    my_hash = JSON.parse(http.body)
+    @users = User.all
+    headers = { exp: Time.now.to_i + 60 } # expire in 60 seconds
```
Comment (serg-kovalev) [inline app.rb]: I'd recommend move expiration time to constant

**Pull request closed without merging**